src/re.rs

```diff
@@ -159,6 +159,14 @@ impl fmt::Debug for Regex {
     }
 }
 
+impl PartialEq for Regex {
```

Comment:
@yorhel Could you please put a doc comment on this impl explaining the definition of equality used? (Something like what you said in your initial comment is just fine.)